lib/Mojo/IOLoop/Subprocess.pm

```diff
@@ -10,6 +10,7 @@ use Storable;
 has deserialize => sub { \&Storable::thaw };
 has ioloop      => sub { Mojo::IOLoop->singleton }, weak => 1;
 has serialize   => sub { \&Storable::freeze };
+has _progress_writer => undef, weak => 1;
```

Comment:
We don't use private attribute accessors. Why do you need to weaken the reference?

Reply:
I weakened the reference thinking that would allow the pipe to disappear once `->run` completes (but before `$subprocess` goes out of scope, and thus save on resources. Is there something wrong with that thinking?